evalml/automl/automl_search.py

```diff
@@ -624,8 +648,48 @@ def __init__(
         self._best_pipeline = None
         self._searched = False
 
-        self.X_train = infer_feature_types(X_train)
-        self.y_train = infer_feature_types(y_train)
+        if self.holdout_set_size < 0:
+            raise ValueError(
+                "Holdout set size must be greater than 0. Set holdout set size to 0 to disable holdout set evaluation.",
+            )
+        if (
```

Comment:
I had a bit of trouble understanding what each case here does, could you add some comments to make it easier to read?